code/game/objects/items/reagent_containers/food/drinks/drinkingglass.dm

```diff
@@ -632,6 +632,70 @@
 			center_of_mass = "x=16;y=10"
 			return
 
+/obj/item/reagent_container/food/drinks/drinkingglass/attack(mob/living/target as mob, mob/living/user as mob)
+	if(!target)
+		return
+
+	if(user.a_intent != INTENT_HARM)
+		return ..()
+
+	force = 15
+
+	var/obj/limb/affecting = user.zone_selected
+	var/drowsy_threshold = 0
+
+	drowsy_threshold = CLOTHING_ARMOR_MEDIUM - target.getarmor(affecting, ARMOR_MELEE)
+
+	target.apply_damage(force, BRUTE, affecting, sharp=0)
+
+	if(affecting == "head" && iscarbon(target) && !isXeno(target))
+		for(var/mob/O in viewers(user, null))
+			if(target != user) O.show_message(text(SPAN_DANGER("<B>[target] has been hit over the head with a glass of [name], by [user]!</B>")), 1)
+			else O.show_message(text(SPAN_DANGER("<B>[target] hit \himself with a glass of [name] on the head!</B>")), 1)
```

Comment:
this is forbidden, put them on new lines
```suggestion
			if(target != user)
				O.show_message(text(SPAN_DANGER("<B>[target] has been hit over the head with a glass of [name], by [user]!</B>")), 1)
			else
				O.show_message(text(SPAN_DANGER("<B>[target] hit \himself with a glass of [name] on the head!</B>")), 1)
```

Reply:
done